### PR TITLE
Use cargo workspaces for publish

### PR DESCRIPTION
**Pull request by Rigidity**

### CODE REVIEW

File: .github/workflows/build-crate-and-npm.yml [hunk 1]
```diff
@@ -54,122 +54,13 @@ jobs:
...
-    - name: publish to crates.io if tagged (chia)
-      continue-on-error: true
+    - name: publish to crates.io if tagged
       if: startsWith(github.event.ref, 'refs/tags')
       env:
         CARGO_REGISTRY_TOKEN: ${{ secrets.cargo_registry_token }}
       run: |
-        cargo publish
+        cargo install cargo-workspaces --locked
+        cargo ws publish --publish-as-is --allow-dirty
```
Comment (arvidn) [inline .github/workflows/build-crate-and-npm.yml:63]: I seem to recall that `--allow-dirty` is risky. Does it not include the dirty files in the package?

File: .github/workflows/build-crate-and-npm.yml [hunk 2]
```diff
@@ -54,122 +54,13 @@ jobs:
...
-    - name: publish to crates.io if tagged (chia)
-      continue-on-error: true
+    - name: publish to crates.io if tagged
       if: startsWith(github.event.ref, 'refs/tags')
       env:
         CARGO_REGISTRY_TOKEN: ${{ secrets.cargo_registry_token }}
       run: |
-        cargo publish
+        cargo install cargo-workspaces --locked
+        cargo ws publish --publish-as-is
```
Comment (arvidn) [inline .github/workflows/build-crate-and-npm.yml:63]: the only thing the readme has to offer about this option is: "Publish crates from the current commit without versioning". What does that mean? Surely the crates that are published will have versions.

**Comment by arvidn** [comment]
this is a good improvement

File: .github/workflows/build-crate-and-npm.yml (hunk 2 above)
Comment (Rigidity) [reply .github/workflows/build-crate-and-npm.yml:63]: Confusing wording, it just means it wont bump version automatically since we already have

**Review by arvidn — APPROVED** [review]

**Pull request merged**